Adds the Supermatter Crystal and supermatter monitoring console

Mackazonia98 opened the pull request:
## About the PR
<!-- What did you change? -->
<!-- If this is a code change, summarize at high level how your new code works. This makes it easier to review. -->
This PR adds the supermatter crystal and associated monitoring console, including the machine board required to build it.

## Why / Balance
<!-- Discuss how this would affect game balance or explain why it was changed. Link any relevant discussions or issues. -->
The Supermatter is often requested by engineering-inclined players. This PR adds the supermatter to expand the number of gameplay options available at Edison and give brave engineers something new to experiment with.

## Technical details
<!-- Summary of code changes for easier review. -->
- Ports supermatter components, systems, audio files, textures, and prototypes from Goob/Funky. A majority of changes are new file additions. No mapping changes.
- Adjustments to some radio/intercom components to allow the SM to complain over comms.
- Minor additions to the gravity well systems for SM specific gravity pulses.
- Additions to the beam targeting system to allow for more dynamic lightning behavior.
## How to test
<!-- Describe a procedure to test this feature, along with expected output/behavior. -->
Spawn in a supermatter crystal and the relevant monitoring console. Throw things into it for kicks, or create an actual atmospherics setup to see how it reacts. Delaminate it for explosions. Tesla and Singularity will not form away from Edison and retain their inbuilt safety. Harvest plasma and o2 from the angry yellow rock.

## Media
<!-- Attach media if the PR makes ingame changes (clothing, items, features, etc).
Small fixes/refactors are exempt. Media may be used in SS14 progress reports with credit. -->
<img width="819" height="744" alt="509212544-5ae81440-61fc-4a8d-9a63-ea0a89559232" src="https://github.com/user-attachments/assets/1fa6e1cb-2067-4fd0-8197-b1142657cef5" />
<img width="1053" height="573" alt="509212551-34040c92-cd23-4844-a4f0-cb5714595073" src="https://github.com/user-attachments/assets/3968d27d-63ed-4926-b388-d1c852cb5da0" />
<img width="667" height="685" alt="509212553-bf0f5044-a63a-4720-b1a8-fd198b10b412" src="https://github.com/user-attachments/assets/29c0fdd5-d316-4086-a145-9308c31dbddf" />
<img width="440" height="95" alt="509212556-d68473fa-3cc3-46eb-9ed6-0a54e3df4723" src="https://github.com/user-attachments/assets/89709f33-c29a-40fe-88a0-f357c13890cc" />
<img width="901" height="505" alt="Screenshot 2025-11-08 121518" src="https://github.com/user-attachments/assets/bc60601e-4e9e-4563-ba13-1fd357004d18" />


## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [X] I have read [CONTRIBUTING.md](https://github.com/new-frontiers-14/frontier-station-14/blob/master/CONTRIBUTING.md) and and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.
- [X] I have reviewed the [Ship Submission Guidelines](https://frontierstation.wiki.gg/wiki/Ship_Submission_Guidelines) if relevant.
- [x] I confirm that the content in this PR is my own work, and/or is properly attributed to the original author(s).
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them. -->

**Changelog**
<!-- Add a Changelog entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines and take this Changelog template out of the comment block in order for it to show up.
Changelog must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog. -->
:cl:
add: Adds the Supermatter Crystal and SM Monitoring Console.

Mackazonia98 committed:
Adds the supermatter

Adds the supermatter crystal and related systems, including the monitoring console.

Mackazonia98 committed:
Merge branch 'new-frontiers-14:master' into supermatter-fixing

Mackazonia98 committed:
Update SupermatterSystem.Processing.cs

blackknight954 committed:
Merge branch 'master' into supermatter-fixing

blackknight954 committed:
Merge branch 'master' into supermatter-fixing

Mackazonia98 committed:
Fixes SM deleting absorbed gas

The damage handling function of the SM was failing to return gasses used to calculate integrity loss. This is hopefully now fixed.

Mackazonia98 committed:
Adjusts yelltimer to complain less

Doubles the amount of time between SM damage announcements and increases the damage needed to cause warnings from 50 to 100 (The crystal has 900 health.)

Mackazonia98 committed:
Merge branch 'new-frontiers-14:master' into supermatter-fixing

Mackazonia98 committed:
BeamSystem update/SM lightning

Adopts Imp/Funky systems for beam and lightning targeting to increase versatility. Fixes the SM not throwing out random lightning when it should using those new beam components.

Mackazonia98 committed:
Merge branch 'new-frontiers-14:master' into supermatter-fixing

Mackazonia98 committed:
Adds guidebook entry

Adds a guidebook section for the supermatter crystal.

Mackazonia98 commented:
With the implementing of the guidebook entry, the Supermatter crystal itself is effectively finished.
Feedback/developer input will lead to specific values and variables to be changed, such as gas effects, radiation output, and delamination behavior. Through my testing thus far the crystal is fully functional!

The additions to the BeamTargeting and Lightning systems is likely the most noteworthy change that isn't _directly_ related to the supermatter. Default processes haven't been removed, so it shouldn't break anything and serves as more versatility for future contributions involving beams or bolts.

Mackazonia98 committed:
Merge branch 'new-frontiers-14:master' into supermatter-fixing

Mackazonia98 committed:
Fixes singulo/tesla generation

Makes the SM properly spawn a singularity or tesla ball when conditions are met. Prevents the spawned singularity from being level 6 instantly.
Also slightly adjusts lightning chances.

Mackazonia98 committed:
Merge branch 'supermatter-fixing' of https://github.com/Mackazonia98/frontier-station-14 into supermatter-fixing

Mackazonia98 committed:
Merge branch 'new-frontiers-14:master' into supermatter-fixing

Mackazonia98 committed:
Tweaks SM values

Small adjustments to some atmos values regarding the supermatter, and increasing the delamination timer from thirty seconds to sixty.
Helium: PowerTransmit 0.1<0, HeatResistance 0.1<1
Frezon: HeatResistance 1<2
Tritium: PowerMixRatio 1<1.092

Mackazonia98 committed:
Merge branch 'new-frontiers-14:master' into supermatter-fixing

Mackazonia98 committed:
Merge branch 'new-frontiers-14:master' into supermatter-fixing

Mackazonia98 committed:
Merge branch 'new-frontiers-14:master' into supermatter-fixing

Mackazonia98 committed:
Lighting changes

Allows the supermatter to change the color of light it emits based on integrity.
Small commenting additions for readability.

Mackazonia98 committed:
Merge branch 'supermatter-fixing' of https://github.com/Mackazonia98/frontier-station-14 into supermatter-fixing

Mackazonia98 committed:
Merge branch 'new-frontiers-14:master' into supermatter-fixing

Mackazonia98 committed:
Merge branch 'new-frontiers-14:master' into supermatter-fixing

blackknight954 commented:
Apart from the above comment, though the YML portions seem alright, the SM itself seems to at least function, although I can admit I'm not super familiar with all its intricacies. I just read the guidebook entry for it.

Mackazonia98 committed:
Deletes unneeded file

Removes the autoclave yml that hitchhiked in the machines folder.

Mackazonia98 committed:
Merge branch 'supermatter-fixing' of https://github.com/Mackazonia98/frontier-station-14 into supermatter-fixing

Mackazonia98 committed:
Merge branch 'new-frontiers-14:master' into supermatter-fixing

Mackazonia98 committed:
Merge branch 'new-frontiers-14:master' into supermatter-fixing

blackknight954 reviewed (approved):
Played with it a little bit and it works. It starts to snowball pretty hard towards delamming once the room lights on fire though, even when spaced, but that's likely just user error anyway.

Mackazonia98 committed:
Makes AGhosts immune to ashing

Hilarious, but not intentional.

Mackazonia98 committed:
Merge branch 'new-frontiers-14:master' into supermatter-fixing

blackknight954 reviewed (approved):
Aghosts no longer get ashed from the SM, apart from that it functions the same, so here's another approval

Alkheemist commented on Content.Server/Lightning/LightningSystem.cs:
```diff
@@ -59,53 +78,166 @@ public void ShootLightning(EntityUid user, EntityUid target, string lightningPro
-        //TODO: add support to different priority target tablem for different lightning types
-        //TODO: Remove Hardcode LightningTargetComponent (this should be a parameter of the SharedLightningComponent)
-        //TODO: This is still pretty bad for perf but better than before and at least it doesn't re-allocate
+        //To Do: add support to different priority target tablem for different lightning types
```
why is this changed? Going to trip someone up later when they search for `todo` and miss these

Alkheemist commented on Resources/Audio/Announcements/supermatter.ogg:
should go in a namespace folder

Alkheemist commented on Resources/Locale/en-US/guidebook/guides.ftl:
```diff
@@ -54,6 +54,7 @@ guide-entry-ame = Antimatter Engine (AME)
 guide-entry-singularityteslaengine = Singularity / Tesla
 guide-entry-singularityengine = Singularity Engine
 guide-entry-teslaengine = Tesla Engine
+guide-entry-sm = Supermatter Engine
```
namespace comments or (ideally) move to a namespaced guides.ftl file.

Alkheemist commented on Resources/Prototypes/Entities/Mobs/Player/admin_ghost.yml:
```diff
@@ -106,6 +106,7 @@
     stealthy: true
   - type: Stripping
   - type: SolutionScanner
+  - type: SupermatterImmune
```
namespace comments

Alkheemist commented on Resources/Prototypes/Entities/Structures/Power/Generation/Singularity/singularity.yml:
```diff
@@ -109,6 +109,7 @@
             sprite: Structures/Power/Generation/Singularity/singularity_6.rsi
             state: singularity_6
             scale: .9,.9
+  - type: SupermatterImmune #SM Port, stops SM from ashing it
```
namespace comments

Alkheemist reviewed (changes requested):
still needs some work on the comments.

Mackazonia98 committed:
Commenting updates

Improves namespace commenting/placement
Removes unnecessary intercom/ai components related to TTS
Unifies some formatting

Mackazonia98 commented:
Currently, I believe the sticking point is logistical issues on what to actually do with the supermatter once it's available.
Though with the namespace comments (hopefully) being satisfactory, this PR doesn't have much left to be added. Mapping is the next step, though I'm a bit stuck on what to do about it!
I've been heavily distracted as of late, but I'm planning to get up to speed and figure out what needs to be done with our favorite angry crystal.

DevArchwave commented:
Oh! @Mackazonia98, there's a single commit in the PR right above my comment that fixes the console not updating, if you're looking to just quickly fix things instead of porting all the imp updates just yet. We've been testing it right now and we just had a major Edison remap to fit it among many other things.

rebaserHEAD commented:
I believe Goob or Funky had a similar issue with a rat spawning underneath it. I think they have a fix for it somewhere.

DevArchwave commented:
Oh, right, I forgot to mention, we have a companion PR that entirely revamps Edison to fill more of a hybrid shipyard role, if you wish to have a look! There are a few more things we changed along the way, but maybe it could serve as some inspiration? I do think the work that was done there was lovely.
https://github.com/ARF-SS13/coyote-frontier/pull/961
